internal/cmd/testdata/petstore3/provider_code_spec.json

```diff
@@ -149,6 +149,86 @@
 								}
 							]
 						}
+					},
+					{
+						"name": "pets",
+						"list_nested": {
```

Comment:
One awkwardness with plural data sources is that APIs _generally_ don't guarantee ordering of results, so providers will use a set instead of a list to prevent practitioners from potentially depending on any ordering behaviors. We may find that a lot of OAS does not declare this Terraform important implementation detail via the set format, so it may be somewhat preferable to default to a set and instead require opt-in to responses actually being ordered.

Reply:
I was hoping we could keep using the shared `schema.BuildDataSourceAttribute(name, schema.Computed), since it already is being used externally (for query/path parameters) 🤔

So the set mapping might need to be extracted from the collection mapping so we can use it directly to achieve this.

I also could just apply the `set` format before mapping 😄, which is hackish but still expressive of what we want lol
```go
readResponseAttributes := attrmapper.DataSourceAttributes{}
if readResponseSchema.Type == util.OAS_type_array {
	logger.Debug(fmt.Sprintf("response body is an array, building '%s' collection attribute", name))

	// Default it to a set!
	readResponseSchema.Format = util.TF_format_set

	collectionAttribute, schemaErr := readResponseSchema.BuildDataSourceAttribute(name, schema.Computed)
	if schemaErr != nil {
		return nil, schemaErr
	}

	readResponseAttributes = append(readResponseAttributes, collectionAttribute)
} else {
	attributes, schemaErr := readResponseSchema.BuildDataSourceAttributes()
	if schemaErr != nil {
		return nil, schemaErr
	}

	readResponseAttributes = attributes
}
```

Reply:
We could also make this configurable in the generator config (along with eventually the name?)

Reply:
I made the former change, we can add in additional configuration later if needed